runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/KeyedTimerData.java

```diff
@@ -192,7 +193,10 @@ public void encode(KeyedTimerData<K> value, OutputStream outStream)
         keyBytes = baos.toByteArray();
       }
 
-      return new KeyedTimerData(keyBytes, key, timer);
+      final String timerFamilyId = inStream.available() > 0 ? STRING_CODER.decode(inStream) : "";
```

Comment:
Can you add a comment saying all new fields must be append to the end in order not to break forward/backward compatibility?

Reply:
I will add the comment when moving the recently added outputTimestamps field.